src/main/java/org/elasticsearch/search/aggregations/AggregationPhase.java

```diff
@@ -120,6 +122,7 @@ public void execute(SearchContext context) throws ElasticsearchException {
                 query = new FilteredQuery(query, searchFilter);
             }
             try {
+                collector.preCollection();
```

Comment:
This is a different set of aggregators (the `global` ones) which run separately from regular aggregators. Regular aggregators are pre-collected in preProcess but global aggregators do everything in execute (they run on a MatchAllDocsQuery).